test/libyul/yulSyntaxTests/eof/extcalls.yul

```diff
@@ -0,0 +1,25 @@
+object "a" {
+    code {
+        if extcall(address(), 0, 0) {
+            return(0, 32)
+        }
+
+        if extdelegatecall(address(), 0, 0, 10) {
+            return(0, 32)
+        }
+
+        if extstaticcall(address(), 0, 0, 10) {
+            return(0, 32)
+        }
+
+        revert(0, 0)
+    }
+}
+
+// ====
+// EVMVersion: >=prague
+// bytecodeFormat: >=EOFv1
+// ----
+// TypeError 7000: (35-42): Function "extcall" expects 4 arguments but got 3.
+// TypeError 7000: (110-125): Function "extdelegatecall" expects 3 arguments but got 4.
+// TypeError 7000: (197-210): Function "extstaticcall" expects 3 arguments but got 4.
```

Comment:
Is the test passing in the wrong number of arguments on purpose? If so, why? Shouldn't it be testing that these builtins compile without errors?

Reply:
BTW, since testing `if`s and `revert` is not the purpose of the test, and we don't actually execute the code, you could make it much more concise:

```yul
pop(extcall(address(), 0, 0))
pop(extdelegatecall(address(), 0, 0, 10))
pop(extstaticcall(address(), 0, 0, 10))
```

Same applies to the other tests. Possibly using `sstore()` if `pop()` gets optimized out.

Reply:
done